File: app/src/main/java/com/example/mvctutorial/designpattern/_04factory/DefaultItem.java
package com.example.mvctutorial.designpattern._04factory;


import com.example.mvctutorial.designpattern._04factory.fw.Item;

public class DefaultItem extends Item {

    private String itemName;

    public DefaultItem(String itemName) {
        this.itemName = itemName;
    }

    @Override
    public void use() {
        System.out.println(itemName +" 사용했습니다!");
    }


}

File: app/src/main/java/com/example/mvctutorial/designpattern/_04factory/DefaultItem.kt
package com.example.mvctutorial.designpattern._04factory

import com.example.mvctutorial.designpattern._04factory.fw.Item

class DefaultItem(private val itemName: String) : Item() {


    override fun use() {
        println("$itemName 사용했습니다!")
    }


}